zephyr-apps/mppt/src/adc.h

```diff
@@ -0,0 +1,26 @@
+#include <zephyr/zephyr.h>
+#include <zephyr/drivers/adc.h>
+
+// Find devicetree configurations
+#if !DT_NODE_EXISTS(DT_PATH(zephyr_user)) || \
+	!DT_NODE_HAS_PROP(DT_PATH(zephyr_user), io_channels)
+#error "No suitable devicetree overlay specified"
+#endif
+
+#define DT_SPEC_AND_COMMA(node_id, prop, idx) \
+	ADC_DT_SPEC_GET_BY_IDX(node_id, idx),
+
+#define LABEL_AND_COMMA(node_id, prop, idx) \
+	DT_LABEL(DT_IO_CHANNELS_CTLR_BY_IDX(node_id, idx)),
+
+// ADC Settings Todo: Move this to the overlay file
```

Comment:
do todo?

Reply:
This is not super important, and I've already sunk a fair bit of time dealing with the overlay file config, so I rather work on the higher priority stuff first.